OmsAgent/omsagent.py

```diff
@@ -46,100 +49,279 @@
+            continue
+
+        # Check if this VM distribution version is supported
+        vm_ver_split = vm_ver.split('.')
+        for supported_ver in supported_dists[supported_dist]:
+            supported_ver_split = supported_ver.split('.')
+
+            vm_ver_match = True
+            for idx, supported_ver_num in enumerate(supported_ver_split):
+                try:
+                    vm_ver_num = vm_ver_split[idx]
+                except IndexError:
+                    vm_ver_match = False
+                    break
+                if vm_ver_num is not supported_ver_num:
+                    vm_ver_match = False
+                    break
+            if vm_ver_match:
+                vm_supported = True
+                break
+
+        if vm_supported:
+            break
+
+    return vm_supported
+
+
+def dummy_command(hutil):
+    return 0
+
+
+def run_command_with_retries(hutil, cmd, file_directory, operation, extension_short_name, retries,
+                             initial_sleep_time = 30, sleep_increase_factor = 1):
+    '''
+    Some commands fail because the package manager is locked (apt-get/dpkg only); this will
+      allow retries on failing commands.
+    Logic used: will retry up to retries times with initial_sleep_time in between tries
+    Note: install operation times out from WAAgent at 15 minutes, so do not wait longer.
+    '''
+    try_count = 0
+    sleep_time = initial_sleep_time # seconds
+    while try_count <= retries:
+        exit_code = ScriptUtil.run_command(hutil, ScriptUtil.parse_args(cmd), file_directory,
+                                           operation, extension_short_name,
+                                           hutil.get_extension_version())
+        if exit_code is 0:
+            break
+        try_count += 1
+        hutil.log('Retrying command ' + cmd + ' because it failed with exit code ' + str(exit_code))
+        time.sleep(sleep_time)
+        sleep_time *= sleep_increase_factor
+
+    return exit_code
+
+
+def log_and_exit(hutil, operation, exit_code = 1, message = ''):
+    if exit_code is 0:
+        waagent.Log(message)
+        hutil.log(message)
+        hutil.do_exit(exit_code, operation, 'success', str(exit_code), message)
+    else:
+        waagent.Error(message)
+        hutil.error(message)
+        hutil.do_exit(exit_code, operation, 'failed', str(exit_code), message)
 
 
 def install(hutil):
-    file_name = getbundlefilename();
     file_directory = os.path.join(os.getcwd(), PackagesDirectory)
-    file_path = os.path.join(file_directory, file_name)
+    file_path = os.path.join(file_directory, BundleFileName)
 
     os.chmod(file_path, 100)
-    cmd = InstallCommandTemplate.format(file_name)
-    ScriptUtil.run_command(hutil, ScriptUtil.parse_args(cmd), file_directory, 'Install', ExtensionShortName, hutil.get_extension_version())
+    cmd = InstallCommandTemplate.format(BundleFileName)
+    waagent.Log("Starting command %s --upgrade." %(BundleFileName))
```

Comment:
Minor: I'd change the string arg with .format() and also with cmd, rather than with BundleFileName. Same comment applies in uninstall().